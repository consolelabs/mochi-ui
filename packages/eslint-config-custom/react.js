const { resolve } = require('node:path')

const project = resolve(process.cwd(), 'tsconfig.json')

const dfLintConfig = require.resolve('@dwarvesf/react-eslint-config')

module.exports = {
  extends: [dfLintConfig, 'eslint-config-turbo'],
  plugins: ['prettier'],
  parserOptions: {
    project,
  },
  globals: {
    JSX: true,
  },
  ignorePatterns: [
    'node_modules/',
    'dist/',
    '.eslintrc.js',
    '**/*.css',
    '*.snap',
  ],
  settings: {
    'import/resolver': {
      typescript: {
        project,
      },
      node: {
        extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
  rules: {
    'react/function-component-definition': 'off',
    'turbo/no-undeclared-env-vars': 'off',
    'react/jsx-key': 'off',
    'no-dupe-class-members': 'off',
    'prefer-arrow-callback': 'off',
    'import/export': 'off',
    'prettier/prettier': [
      'error',
      {
        bracketSpacing: true,
        jsxBracketSameLine: false,
        jsxSingleQuote: false,
        printWidth: 80,
        proseWrap: 'always',
        semi: false,
        singleQuote: true,
        tabWidth: 2,
        trailingComma: 'all',
        xmlWhitespaceSensitivity: 'ignore',
      },
    ],
  },
  overrides: [
    {
      parser: '@typescript-eslint/parser',
      files: ['**/*.ts', '**/*.tsx'],
      plugins: ['@typescript-eslint'],
      rules: {
        'no-use-before-define': 'off',
        'react/react-in-jsx-scope': 'off',
        'react/jsx-sort-props': 'off',
        'no-void': 'off',
        'no-unused-vars': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            argsIgnorePattern: '^_',
            varsIgnorePattern: '^_',
            caughtErrorsIgnorePattern: '^_',
          },
        ],
        '@typescript-eslint/no-use-before-define': [
          'error',
          { functions: false, classes: false },
        ],
      },
    },
  ],
}
