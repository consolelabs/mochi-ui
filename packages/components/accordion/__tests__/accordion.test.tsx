import { render, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import {
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionTrigger,
} from '../src/accordion'

const mockedData = [
  {
    value: '1',
    triggerLabel: 'Item 1',
    contentLabel: 'Content 1',
  },
  {
    value: '2',
    triggerLabel: 'Item 2',
    contentLabel: 'Content 2',
  },
]

describe('Accordion', () => {
  it('expands and collapses the accordion item when clicked', async () => {
    const { getByTestId } = render(
      <Accordion type="single">
        {mockedData.map((item) => (
          <AccordionItem key={item.value} value={item.value}>
            <AccordionTrigger data-testid={item.triggerLabel}>
              {item.triggerLabel}
            </AccordionTrigger>
            <AccordionContent>{item.contentLabel}</AccordionContent>
          </AccordionItem>
        ))}
      </Accordion>,
    )

    const item1Trigger = getByTestId('Item 1')
    const item2Trigger = getByTestId('Item 2')

    expect(item1Trigger).toHaveAttribute('aria-expanded', 'false')
    expect(item2Trigger).toHaveAttribute('aria-expanded', 'false')

    expect(item1Trigger).toBeInTheDocument()
    expect(item2Trigger).toBeInTheDocument()

    userEvent.click(item1Trigger)
    await waitFor(() => {
      expect(item1Trigger).toHaveAttribute('aria-expanded', 'true')
    })

    userEvent.click(item2Trigger)
    await waitFor(() => {
      expect(item2Trigger).toHaveAttribute('aria-expanded', 'true')
    })

    userEvent.click(item2Trigger)
    await waitFor(() => {
      expect(item1Trigger).toHaveAttribute('aria-expanded', 'false')
    })
  })
})
