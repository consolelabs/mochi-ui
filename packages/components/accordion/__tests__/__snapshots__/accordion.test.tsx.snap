// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders correctly 1`] = `
<div>
  <div
    class="p-2 space-y-1 rounded-lg shadow-md"
    data-orientation="vertical"
  >
    <div
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r1:"
          aria-expanded="false"
          class="flex-1 flex font-medium text-sm items-center justify-between [&[data-state=open]>#chevron]:rotate-180 gap-3 p-2"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          <span
            class="flex flex-1"
          >
            Item 1
          </span>
          <span
            class="text-neutral-500 shrink-0 transition-transform duration-200 text-base"
            id="chevron"
          >
            <svg
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 0 1 1.06.02L10 11.168l3.71-3.938a.75.75 0 1 1 1.08 1.04l-4.25 4.5a.75.75 0 0 1-1.08 0l-4.25-4.5a.75.75 0 0 1 .02-1.06Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r0:"
        class="text-sm"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r1:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="false"
          class="flex-1 flex font-medium text-sm items-center justify-between [&[data-state=open]>#chevron]:rotate-180 gap-3 p-2"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r2:"
          type="button"
        >
          <span
            class="flex flex-1"
          >
            Item 2
          </span>
          <span
            class="text-neutral-500 shrink-0 transition-transform duration-200 text-base"
            id="chevron"
          >
            <svg
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 0 1 1.06.02L10 11.168l3.71-3.938a.75.75 0 1 1 1.08 1.04l-4.25 4.5a.75.75 0 0 1-1.08 0l-4.25-4.5a.75.75 0 0 1 .02-1.06Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r2:"
        class="text-sm"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r3:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;
